packages/transaction-controller/src/TransactionController.ts

```diff
@@ -570,12 +582,21 @@ export class TransactionController extends BaseController<
    * Attempts to speed up a transaction increasing transaction gasPrice by ten percent.
    *
    * @param transactionID - The ID of the transaction to speed up.
-   * @param gasValues - The gas values to use for the speed up transation.
+   * @param gasValues - The gas values to use for the speed up transaction.
+   * @param opts - Additional options to control how the transaction is speed up.
+   * @param opts.actionId - Unique ID to prevent duplicate requests
    */
   async speedUpTransaction(
     transactionID: string,
     gasValues?: GasPriceValue | FeeMarketEIP1559Values,
+    opts: { actionId?: string } = {},
   ) {
+    const { actionId } = opts;
+    // If transaction is found for same action id, do not create a new speed up transaction.
+    if (this.getTransactionWithActionId(actionId)) {
```

Comment:
Do we not need identical logic on `stopTransaction` too?

Reply:
I was confused by adding this logic to `stopTransaction`maily because it does not include the `rawTransaction` into the state so the second time calling it it will create the transaction anyway because in the first call the `actionId` was not saved. I will ping you so we can double-check this point together.